pkg/policy/api/rule.go

```diff
@@ -61,3 +61,29 @@ type Rule struct {
 	// +optional
 	Description string `json:"description,omitempty"`
 }
+
+// RequiresDerivative it return true if the rule has a child rule.
+func (r *Rule) RequiresDerivative() bool {
+	for _, rule := range r.Egress {
+		if rule.RequiresDerivative() {
+			return true
+		}
+	}
+	return false
+}
+
+// CreateDerivative will return a new Rule with the new data based gather
+// by the rules that autogenerated new Rule
+func (r *Rule) CreateDerivative() (*Rule, error) {
+	newRule := r.DeepCopy()
+	newRule.Egress = []EgressRule{}
```

Comment:
The deepcopy should already have made these for us. I think this is probably fine as it is but an alternative is to keep the egressrules array here, but replace each element with the `CreateDerivative` version as you iterate (you would need to use the index). This would mean that we never touch rules that don't have a groups field. EgressRule.CreateDerivative() already does that but it isn't super clear (I guess it technically makes a copy but the outcome is the same).

Reply:
But if the copy is not in place, how can I loop over the old one? I think that is the simplest way, or I can replace at the end, but I think that it's clear on this way.